### PR TITLE
WIP - #3071 / Added Grants Contributions to activity report

### LINKED ISSUE #3071: add grants to activity_report.py
report all the grant contributions in activity_report.py similar to tips/bounties/kudos

### PR DESCRIPTION
**Pull request by kuhnchris**
##### Description
As discussed in #3071 - adding Grants to the activity report.

##### Fixes

Fixes #3071

### CODE REVIEW

File: app/dashboard/management/commands/activity_report.py [hunk 1]
```diff
@@ -212,13 +239,25 @@ def handle(self, *args, **options):
             created_on__lte=options['end_date']
         ).order_by('created_on', 'id')
         formted_enssubreg = imap(self.format_ens_reg, enssubregistrations)
-
+        
+        #grants contribution objcets
+        grantcontobjs = Contribution.objects.filter(
+            created_on__gte=options['start_date'],
+            created_on__lte=options['end_date']
```
Comment (owocki) [inline app/dashboard/management/commands/activity_report.py]: maybe filter for contributions on the mainnet ?

**Comment by owocki** [comment]
LGTM; other than the stickler comments and the dead code

File: app/dashboard/management/commands/activity_report.py (hunk 1 above)
Comment (kuhnchris) [reply app/dashboard/management/commands/activity_report.py]: If the suggestion works for you for now I'll clean it up and get it rolling. 🚢

File: app/dashboard/management/commands/activity_report.py [hunk 2]
```diff
@@ -172,6 +173,29 @@ def format_ens_reg(self, ensreg):
...
+            'denomination': contrib.subscription.token_symbol,
+            'amount_eth': convert_amount(contr_sub.amount_per_period, contr_sub.token_symbol, 'ETH'),
+            'amount_usdt': convert_amount(contr_sub.amount_per_period, contr_sub.token_symbol, 'USD'),
+            'from_address': contrib.subscription.contributor_address,
+            'claimee_address': contrib.subscription.grant.contract_address,
+            'repo': 'n/a',
+            'from_username': contrib.subscription.contributor_profile.handle,
+            'fulfiller_github_username': contr_prof.handle if contr_prof else "",
+            'status': 'sent',
+            'comments': f"Gas: {contrib.gas_price}, Nonce: {contrib.nonce}, Grant: {contr_sub.grant.id}, Subscription: {contr_sub.id}, TX: {contrib.tx_id}",
```
[the remark this answers is not in the document]
Comment (kuhnchris) [reply app/dashboard/management/commands/activity_report.py]: @owocki anything I can do here? want me to build the string piece by piece?

**Comment by kuhnchris** [comment]
@frankchen07 @PixelantDesign @owocki any reason this is still out here, shall we bin it?

**Comment by owocki** [comment]
sorry for letting this drop.   is the current code working / tested?

File: app/dashboard/management/commands/activity_report.py (hunk 1 above)
Comment (owocki) [reply app/dashboard/management/commands/activity_report.py]: yes pls

**Comment by kuhnchris** [comment]
Last time I remember it was tested, but that's a good half-year old, so, I'd prefer to have a retest of this? Hence my question if it's still neccessary or already implemented elsewhere

**Comment by frankchen07** [comment]
took a look at the grants page - it looks like there's activity under the activity tab on every grant - I'm assuming this is implemented already?

**Comment by kuhnchris** [comment]
Then let's close it.

**Pull request closed without merging**